### PR TITLE
testing/grass: add GRASS GIS

### PR DESCRIPTION
**Pull request by zarch**
The package it is almost working. :-)
I'm able to `abuild -r`, but I have still an error:
```
[...]
>>> grass*: Tracing dependencies...
>>> ERROR: grass*: libgrass_arraystats.7.8.so: path not found
        attr
        bash
        [...]
```
Do you know how can I fix this?
However, the process it seems to continue until the end:
```
[...]
>>> grass*: Package size: 81.8 MB
>>> grass*: Compressing data...
>>> grass*: Create checksum...
>>> grass*: Create grass-7.8.0-r0.apk
>>> grass: Build complete at Thu, 26 Sep 2019 13:39:28 +0000 elapsed time 0h 9m 56s
>>> grass: Cleaning up srcdir
>>> grass: Cleaning up pkgdir
>>> grass: Uninstalling dependencies...
(1/1) Purging .makedepends-grass (20190926.132935)
OK: 885 MiB in 324 packages
>>> grass: Updating the testing/x86_64 repository index...
>>> grass: Signing the index...
```

And there are also few warnings:
```
>>> grass-gui*: Running split function gui...
>>> grass-gui*: Preparing subpackage grass-gui...
>>> grass-gui*: Stripping binaries
>>> WARNING: grass-gui*: No arch specific binaries found so arch should probably be set to "noarch"
>>> grass-gui*: Running postcheck for grass-gui
>>> grass-fonts*: Running split function fonts...
>>> grass-fonts*: Preparing subpackage grass-fonts...
>>> grass-fonts*: Stripping binaries
>>> WARNING: grass-fonts*: No arch specific binaries found so arch should probably be set to "noarch"
>>> grass-fonts*: Running postcheck for grass-fonts
>>> grass-demolocation*: Running split function demo...
>>> grass-demolocation*: Preparing subpackage grass-demolocation...
>>> grass-demolocation*: Stripping binaries
>>> WARNING: grass-demolocation*: No arch specific binaries found so arch should probably be set to "noarch"
```
Any reference on how o fix this? 
- `grass-gui` contains the wxpython GUI, so basically some python files, and also some `.png` files for the icons of the GUI;
- `grass-fonts` contains the following files:
```
cyrilc.hmp   gothgbt.hmp  gothitt.hmp  greekcs.hmp  greeks.hmp  hersh.oc2  hersh.oc4    italiccs.hmp  romanc.hmp   romand.hmp  romant.hmp   scripts.hmp
fonts.table  gothgrt.hmp  greekc.hmp   greekp.hmp   hersh.oc1   hersh.oc3  italicc.hmp  italict.hmp   romancs.hmp  romans.hmp  scriptc.hmp
```
- `grass-demolocation` contains a basic project directory that can be used to test the installation.

How can I set `noarch` only for the sbupackages? I was not able to find a solution on the web.

### CODE REVIEW

File: testing/grass/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	"
+subpackages="
+	$pkgname-dev:dev
+	$pkgname-doc:doc
+	$pkgname-gui:gui
+	$pkgname-fonts:fonts
+	$pkgname-demolocation:demo
+	"
+
+source="${pkgname}-${pkgver}.tar.gz::https://grass.osgeo.org/$pkgname`echo ${pkgver::-2} | tr -d .`/source/$pkgname-$pkgver.tar.gz"
```
Comment (jjsearle) [inline testing/grass/APKBUILD]: You would probably be better adding an additional variable like this to remove the periods
`_pkgver=${pkgver//.}`
You can then just reference your source line like this
`source="$pkgname-$pkgver.tar.gz::https://grass.osgeo.org/grass${_pkgver::-1}/source/grass-$pkgver.tar.gz"`

Note that I have remove the $pkgname in the source url too

**Comment by jjsearle** [comment]
Regarding subpackages, I believe you can set subpackage specific variable inside their respective functions

https://wiki.alpinelinux.org/wiki/Creating_an_Alpine_package#Custom_subpackages

File: testing/grass/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+build() {
+	cd "$builddir"
```
Comment (jjsearle) [inline testing/grass/APKBUILD]: this is not required if you are using the default builddir

**Comment by zarch** [comment]
Hi @jjsearle, 
thank you for reviewing and give me feedbacks!
I did the suggested changes.

> Regarding subpackages, I believe you can set subpackage specific variable inside their respective functions
> 
> https://wiki.alpinelinux.org/wiki/Creating_an_Alpine_package#Custom_subpackages

Ok, I changed the subpackage definition to:
```
	$pkgname-gui:gui:noarch
	$pkgname-fonts:fonts:noarch
	$pkgname-demolocation:demo:noarch
```
And now the warnings disappear.

I still have the error: `>>> ERROR: grass*: libgrass_arraystats.7.8.so: path not found`

Perhaps I can fix this changing the environmental variable: `LD_LIBRARY_PATH`?

**Comment by zarch** [comment]
I see that now the [CI](https://travis-ci.org/alpinelinux/aports/builds/590411669?utm_source=github_status&utm_medium=notification) is failing for the patch:
```
>>> grass: fix-make-install.patch
can't find file to patch at input line 13
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|Reason: We need to define and use the DEST_INST variable when
|executing the `make install`.
|GRASS GGIS developer closed a ticket on this with falg: wontfix.
|See: https://trac.osgeo.org/grass/ticket/1839 
|for further details.
|
|The patch is based on `vince` works:
|https://trac.osgeo.org/grass/attachment/ticket/1839/patch-Install_make.diff
|
|
|--- include/Make/Install.make.1	2019-09-26 13:49:50.551485271 +0200
|+++ include/Make/Install.make	2019-09-26 13:13:34.000000000 +0200
--------------------------
File to patch: 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated
```

when I test it locally I simply confirm the path: `File to patch: include/Make/Install.make`

File: testing/grass/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	"
+subpackages="
+	$pkgname-dev:dev
+	$pkgname-doc:doc
+	$pkgname-gui:gui:noarch
+	$pkgname-fonts:fonts:noarch
+	$pkgname-demolocation:demo:noarch
+	"
+source="
+	${pkgname}-${pkgver}.tar.gz::https://grass.osgeo.org/grass${_pkgver}/source/$pkgname-$pkgver.tar.gz
```
Comment (jjsearle) [inline testing/grass/APKBUILD]: curly braces not required on $pkgname and $pkgver

**Comment by jjsearle** [comment]
Fix your patch by changing it to this
```
--- a/include/Make/Install.make 2019-09-26 13:49:50.551485271 +0200
+++ b/include/Make/Install.make 2019-09-26 13:13:34.000000000 +0200
```

**Comment by jjsearle** [comment]
you also need to add `proj-dev` as a make dependency

**Comment by zarch** [comment]
> I still have the error: `>>> ERROR: grass*: libgrass_arraystats.7.8.so: path not found`

The file is:
```bash
$ find pkg/ -name libgrass_arraystats.7.8.so
pkg/grass/usr/grass78/lib/libgrass_arraystats.7.8.so
```
So I believe I need to fix some environmental variable. I tried with:

```bash
$ LD_LIBRARY_PATH="/src/aports/testing/grass/pkg/grass/usr/grass78/lib:$LD_LIBRARY_PATH" abuild rootpkg
>>> grass: Entering fakeroot...
/src/aports/testing/grass/src/grass-7.8.0/bin.x86_64-pc-linux-musl/grass78
make install-check-built
[...]
>>> grass*: Tracing dependencies...
>>> ERROR: grass*: libgrass_arraystats.7.8.so: path not found
        attr
        bash
[...]
>>> grass*: Package size: 81.8 MB
>>> grass*: Compressing data...
>>> grass*: Create checksum...
>>> grass*: Create grass-7.8.0-r0.apk
```
But the error is still there. Any hints?

**Comment by Cogitri** [comment]
You'll have to disable this on most arches due to the proj-dev dependency: `arch="x86_64 aarch64"`

**Pull request closed without merging**